app/js/components/manageApps/ManageApps.jsx

```diff
@@ -143,64 +147,103 @@ export default class ManageApps extends React.Component {
+          })
+          .catch(e => {
+            resolve(false);
+          });
+      });
 
-    readZippedAddon.then((result) => {
-      if (!result) {
-        this.setState((prevState, props) => {
-          return {
-            displayInvalidZip: true
-          };
-        });
-      } else {
-        this.setState({
-          addonAlreadyInstalled: false,
-        });
+      readZippedAddon.then((result) => {
+        if (!result) {
+          this.setState((prevState, props) => {
+            return {
+              displayInvalidZip: true
+            };
+          });
+        } else {
+          this.setState({
+            addonAlreadyInstalled: false,
+          });
 
-        this.state.appList.map((addon) => {
-          if (addon.name === result.name) {
-            this.setState({
-              addonAlreadyInstalled: true,
-            });
-            const toBeInstalledAddonName = result.name;
-            const installedAddonVersion = parseInt(addon.version);
-            const toBeInstalledAddonVersion = parseInt(result.version);
-            if (installedAddonVersion === toBeInstalledAddonVersion) {
-              this.handleAddonUploadModal(
-                toBeInstalledAddonName,
-                'overwrite',
-                'overwriting');
-            } else if (installedAddonVersion < toBeInstalledAddonVersion) {
-              this.handleAddonUploadModal(
-                toBeInstalledAddonName,
-                'upgrade',
-                'upgrading');
-            } else {
-              this.handleAddonUploadModal(
-                toBeInstalledAddonName,
-                'downgrade',
-                'downgrading');
+          this.state.appList.map((addon) => {
+            if (addon.name === result.name) {
+              this.setState({
+                addonAlreadyInstalled: true,
+              });
+              const toBeInstalledAddonName = result.name;
+              const installedAddonVersion = parseInt(addon.version);
+              const toBeInstalledAddonVersion = parseInt(result.version);
+              if (installedAddonVersion === toBeInstalledAddonVersion) {
+                this.handleAddonUploadModal(
+                  toBeInstalledAddonName,
+                  'overwrite',
+                  'overwriting');
+              } else if (installedAddonVersion < toBeInstalledAddonVersion) {
+                this.handleAddonUploadModal(
+                  toBeInstalledAddonName,
+                  'upgrade',
+                  'upgrading');
+              } else {
+                this.handleAddonUploadModal(
+                  toBeInstalledAddonName,
+                  'downgrade',
+                  'downgrading');
+              }
             }
-          }
-        });
-        this.state.addonAlreadyInstalled === false && this.handleUploadRequest();
+          });
+          this.state.addonAlreadyInstalled === false && this.handleUploadRequest();
```

Comment:
Clarification: What does this line of code do exactly?

Reply:
`this.state.addonAlreadyInstalled === false` short circuits `this.handleUploadRequest`, in that the latter is only run after the expression is true. In this case if an add-on is already installed then the upload request is not sent. That being said, I did not write this code, what has changed in this line is only the indentation 😄

Reply:
👍